src/logging/logging.ts

```diff
@@ -43,20 +35,45 @@ export class Log {
   constructor(private tag_:string) {
     this.logger = freedom.core().getLogger(this.tag_);
   }
+
+  private log_ = (level :string, args :any[]) :void => {
+    var message :string;
+
+    if (2 === args.length && 'string' === typeof(args[0]) && Array.isArray(args[1])) {
```

Comment:
FYI, we use the form 'x == 1' much more than '1 == x' in uProxy code (personally I find it much much easier to read).